### ISSUE TITLE
Non-Softmaxed Classification

### ISSUE THREAD

**Issue by justusschock** [PyTorchLightning/metrics#74]
### Motivation

Support other types for classification metrics. I.e non-softmaxed network outputs. 

For outputs of categorical classification, it does not matter if the output is softmaxed or not. The argmax of these tensors is the same. Can we support those by simply taking an argmax even if the values are out of the 0-1 range?

cc @SkafteNicki Whether we want to support this.

**Comment by SkafteNicki** [comment]
Agree that we should support this. This issue is similar to https://github.com/PyTorchLightning/metrics/issues/60, but we should make it more general than just Accuracy metric.
